RELEASE.md

```diff
@@ -1,33 +1,69 @@
-To release a new version of qtpy on PyPI:
+# Release Procedure
 
-* Close Github milestone
+To release a new version of QtPy on PyPI:
 
-* git fetch upstream && git merge upstream/master
 
-* git clean -xfdi
+## Prepare
 
-* Update CHANGELOG.md using `loghub` to generate the list of issues and PRs merged to add at the top of the file
+* Close Github milestone and ensure all issues are resolved/moved
 
-    loghub -m vX.X.X spyder-ide/qtpy
+* `git switch master && git pull upstream master` (assuming `master` is the release branch)
 
-* Update `_version.py` (set release version, remove 'dev0')
+* Check `MANIFEST.in` and `setup.cfg` to ensure they are up to date
 
-* git add . && git commit -m 'Release X.X.X'
 
-* python setup.py sdist
+## Commit
 
-* python setup.py bdist_wheel
+* `pip install --upgrade loghub`
 
-* twine check dist/*
+* Update `CHANGELOG.md` using `loghub` to generate the list of issues and PRs merged to add at the top of the file:
 
-* twine upload dist/*
+  ```bash
+  loghub -m vX.Y.Z spyder-ide/qtpy
+  ```
 
-* git tag -a vX.X.X -m 'Release X.X.X'
+* Update `__version__` in `__init__.py` (set release version, remove `.dev0`)
 
-* Update `_version.py` (add 'dev0' and increment minor)
+* `git commit -am 'Release X.Y.Z'`
 
-* git add . && git commit -m 'Back to work'
 
-* git push
+## Build
 
-* git push --tags
+**Note**: We use `pip` instead of `conda` here even on Conda installs, to ensure we always get the latest upstream versions of the build dependencies.
+
+* `git clean -xfdi`
+
+* `python -m pip install --upgrade-strategy eager --upgrade build pip setuptools twine wheel`
+
+* `python -bb -X dev -W error -m build`
+
+
+## Test
+
+* `python -bb -X dev -W error -m pip install dist/*.whl`
+
+* `pip check`
+
+* `python -bb -X dev -W error -m pytest qtpy/tests`
```

Comment:
Why this? We don't run our tests locally after creating the wheel and tarball were created.

Reply:
Well, if not, then it risks releasing a broken package to the world, which considering QtPy is used by at least 56 300 other projects on GitHub (not counting Gitlab, Bitbucket, SourceForge, other platforms, local development, implicit deps and those dependabot can't read, which is a lot of them) and is top 200 on PyPI, is not a trivial thing :)